vitals_monitor.cpp

```diff
@@ -0,0 +1,119 @@
+    // Find the alert message for the current vital type and status
+    auto messageIt = messageMap.find(vital.status);
+
+    // Print the message if found
+    if (messageIt != messageMap.end()) {
+        std::cout << messageIt->second << std::endl;
+    }
+}
+
+// Function to print alert based on the vital's status using the selected message map and language
+void printAlert(const VitalData& vital, Language language) {
+    // Find the alert message for the current vital type and status in the selected language
+    printVitalStatus(vital, vitalMessages.at(language).at(vital.type));
+
+    // Check if status is critical and trigger animation
+    if (std::find(criticalStatuses.begin(), criticalStatuses.end(),
+                                            vital.status) != criticalStatuses.end()) {
+        showAlertAnimation(2);
+    }
+}
+
+// Function to convert values to the appropriate unit using vector functions
+double convertToCommonUnit(double value, VitalUnit fromUnit) {
+    // Find the matching conversion rule
+    auto it = std::find_if(conversionRules.begin(), conversionRules.end(),
+                           [fromUnit](const ConversionRule& rule) {
+                               return rule.fromUnit == fromUnit;
+                           });
+
+    // Apply the conversion function if a rule is found
+    return (it != conversionRules.end()) ? it->convertFunc(value) : value;
+}
+
+// Function to check and update the vital status based on the thresholds
+void checkVitalStatus(VitalData* vital) {
+    if (vital == nullptr) return;  // Guard clause for null pointers
+
+    double convertedValue = convertToCommonUnit(vital->value, vital->unit);
+    const VitalLimits& limits = vitalThresholds.at(vital->type);
+
+    // Array of thresholds to map ranges to status
+    std::vector<std::pair<double, VitalStatus>> thresholds = {
+        {limits.low, LOW},
+        {limits.nearLow, NEAR_LOW},
+        {limits.nearHigh, NORMAL},
+        {limits.high, NEAR_HIGH}
+    };
+
+    // Use std::find_if to determine the correct status and update the vital status directly
+    auto statusIt = std::find_if(thresholds.begin(), thresholds.end(),
+                                [convertedValue](const std::pair<double, VitalStatus>& limit) {
+                                    return convertedValue <= limit.first;
+                                });
+
+    if (statusIt != thresholds.end()) {
+        vital->status = statusIt->second;
+    }
+}
+
+// Function to check if all vitals are okay, without using any == or && operators
+bool vitalsOk(std::vector<VitalData>* vitalArray, Language language) {
+    if (vitalArray == nullptr) return true;  // Guard clause for null pointers
+
+    // First, update the status of each vital
+    std::for_each(vitalArray->begin(), vitalArray->end(), [](VitalData& vital) {
+        checkVitalStatus(&vital);  // Pass a pointer to each vital

```

Comment:
Can there be a better name for `checkVitalStatus`? It's actually filling some fields inside `vital`
Something like `fillParamClassification`? Nah... can be something better than that